projects/opentrain/gtfs_pipeline/heatmap/HeatMap.java

```diff
@@ -0,0 +1,500 @@
+    // color1 to color2
+    Color color1 = new Color(255, 208, 235);
+    Color color2 = new Color(126, 0, 67);
+
+    // Normalize data
+    double maxData = 0;
+    double minData = heatMap[10][10];
+    for (int i = 0; i < heatMap.length; i++) {
+      for (int j = 0; j < heatMap[0].length; j++) {
+        if (maxData < heatMap[i][j]) {
+          maxData = heatMap[i][j];
+        }
+        if (heatMap[i][j] != 0 && minData > heatMap[i][j]) {
+          minData = heatMap[i][j];
+        }
+      }
+    }
+
+    // Load image to buffer
+    BufferedImage mapImage = null;
+    try {
+      mapImage = (ImageIO.read(new File(imagePath)));
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+
+    // Create image
+    final BufferedImage image =
+        new BufferedImage(mapImage.getWidth(), mapImage.getHeight(), BufferedImage.TYPE_INT_ARGB);
+    final Graphics2D graphics2D = image.createGraphics();
+
+    graphics2D.drawImage(mapImage, 0, 0, null);
+
+    // Draw the map
+    for (int i = 0; i < heatMap.length; i++) {
+      for (int j = 0; j < heatMap[0].length; j++) {
+        if (heatMap[i][j] == 0) {
+          System.out.println("ERROR:" + "\n" + "heatmap matrix value should not be 0");
+        }
+
+        float value = (float) (heatMap[i][j] - minData) / (float) (maxData - minData);
+
+        int r = (int) (((float) color2.getRed() - color1.getRed()) * value + color1.getRed());
+        int g = (int) (((float) color2.getGreen() - color1.getGreen()) * value + color1.getGreen());
+        int b = (int) (((float) color2.getBlue() - color1.getBlue()) * value + color1.getBlue());
+
+        Color color = new Color(r, g, b, 80);
+        graphics2D.setPaint(color);
+        if (inBorders(i, j)) {
+          graphics2D.fillRect(i, j, 2, 2);
+        }
+      }
+    }
+    for (StopsAround stopAround : stopsAroundList) {
+      graphics2D.setPaint(Color.black);
+      graphics2D.setFont(new Font("TimesRoman", Font.BOLD, 10));
+      int x = getXpixel(stopAround.stop.getStopLat(), stopAround.stop.getStopLon());
+      int y = getYpixel(stopAround.stop.getStopLat(), stopAround.stop.getStopLon());
+
+      graphics2D.drawString("X", x, y);
+    }
+
+    // TODO:Each stop to its place by adding/decreasing to x,y
+    graphics2D.drawString("Y", 330 + 80, 633 + 160);
+    graphics2D.dispose();
+    // Save file
+    boolean createImage = false;
+    try {
+      createImage = ImageIO.write(image, "png", new File(heatMapPath.get()));
+
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    if (!createImage) {
+      System.out.println("can't create image");
+    }
+  }
+
+  public static void main(String[] args) throws IOException {
+    // TODO:
+    //      Boarders of israel,do not paint sea area
+    //      Improve x,y pixel method by using more data
+
+    Flags.parseCurrentPackage(args);
+    StopTime stopTime;
+    Trip trip;
+
+    try (FileInputStream tripInputStream =
```

Comment:
Looks like you're loading the files twice - once here and once inside `loadDataFromFiles()`.
Why not load them only once?

Reply:
I need to return to the start of the file,
i don't find good solution for that.
There is some solution with buffer but if the file is big it doesn't work.
Another problem is if i open the files in main i need to transfer them to `HeatMap` and then i need to make `try` on all the class so which function will close that.
If you know some way to read a file and return to its start point i will improve that.

Reply:
What if you load it all into memory - into a list?

Reply:
If `loadDataFromFiles` is static i can use it without create `HeatMap` object but i can use it on main?

Reply:
Sorry, can you clarify the question?

Reply:
Can you clarify the question?